### ISSUE TITLE
Godot leaks memory when creating `NetworkedMultiplayerENet` client

### ISSUE THREAD

**Issue by qarmin** [godotengine/godot#51196]
### Godot version

3.4.beta.custom_build. 7c1ee0488

### System information

Ubuntu 21.04 - Nvidia GTX 970, Gnome shell 3.38 X11

### Issue description

When executing
```
	var temp_variable = NetworkedMultiplayerENet.new()
	temp_variable.create_client(".", 44, 38, 42, 0)
```
Godot leaks memory
```
Indirect leak of 11056 byte(s) in 1 object(s) allocated from:
    #0 0x7f9fe4d71c47 in __interceptor_malloc ../../../../src/libsanitizer/asan/asan_malloc_linux.cpp:145
    #1 0x6f8bec5 in enet_malloc thirdparty/enet/callbacks.c:40
    #2 0x465f785 in enet_host_create thirdparty/enet/host.c:37
    #3 0x313b522 in NetworkedMultiplayerENet::create_client(String const&, int, int, int, int) modules/enet/networked_multiplayer_enet.cpp:150
    #4 0x317fca8 in MethodBind5R<Error, String const&, int, int, int, int>::call(Object*, Variant const**, int, Variant::CallError&) core/method_bind.gen.inc:4177
    #5 0x11bb651d in Object::call(StringName const&, Variant const**, int, Variant::CallError&) core/object.cpp:918
    #6 0x11e3e88f in Variant::call_ptr(StringName const&, Variant const**, int, Variant*, Variant::CallError&) core/variant_call.cpp:1173
    #7 0x1e36ed7 in GDScriptFunction::call(GDScriptInstance*, Variant const**, int, Variant::CallError&, GDScriptFunction::CallState*) modules/gdscript/gdscript_function.cpp:1034
    #8 0x1c679d2 in GDScriptInstance::call_multilevel(StringName const&, Variant const**, int) modules/gdscript/gdscript.cpp:1166
    #9 0xc3db82b in Node::_notification(int) scene/main/node.cpp:56
    #10 0x1b132bb in Node::_notificationv(int, bool) scene/main/node.h:45
    #11 0x1b15730 in CanvasItem::_notificationv(int, bool) scene/2d/canvas_item.h:162
    #12 0xdfcef9e in Node2D::_notificationv(int, bool) scene/2d/node_2d.h:37
    #13 0x11bb69b7 in Object::notification(int, bool) core/object.cpp:927
    #14 0xc50ed31 in SceneTree::_notify_group_pause(StringName const&, int) scene/main/scene_tree.cpp:973
    #15 0xc4ff2bf in SceneTree::idle(float) scene/main/scene_tree.cpp:528
    #16 0x1966be1 in Main::iteration() main/main.cpp:2140
    #17 0x1840a64 in OS_X11::run() platform/x11/os_x11.cpp:3638
    #18 0x17abd8b in main platform/x11/godot_x11.cpp:55
    #19 0x7f9fe3df4564 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x28564)
```
```
Leaked instance: Reference:1269
Hint: Leaked instances typically happen when nodes are removed from the scene tree (with `remove_child()`) but not freed (with `free()` or `queue_free()`).
Orphan StringName: Reference
StringName: 1 unclaimed string names at exit.
```

### Steps to reproduce

Above

**Comment by akien-mga** [comment]
The `host` should be destroyed with `enet_host_destroy(host);`, which is properly done in `NetworkedMultiplayerENet::close_connection()`. But that's only called if the peer is `active`, which it probably doesn't end up being when created with fuzzer input.

**Comment by akien-mga** [comment]
Fixed by #51200.
